src/classes/ALLO_PaymentSync_TEST.cls

```diff
@@ -221,14 +221,143 @@ private with sharing  class ALLO_PaymentSync_TEST {
+        System.assertEquals(null, queryAllocations[1].Opportunity__c, 'Only Payment allocations should have been created');
+        System.assertEquals(null, queryAllocations[2].Opportunity__c, 'Only Payment allocations should have been created');
+
+        //original payment allocations
+        assertAllocation(queryAllocations[0], 200, defaultGau.Id, false);
+        assertAllocation(queryAllocations[1], 300, gaus[1].Id, false);
+        assertAllocation(queryAllocations[2], 500, gaus[0].Id, false);
+    }
+
+
+    /*******************************************************************************************************
+    * @description When changing the amount of Opportunity allocations connected to paid payments
+    * Payment Allocations for paid payments will not be changed
+    ********************************************************************************************************/
+    @isTest
+    static void doNotUpdatePaidPaymentsWithOpportunityChanges() {
+        makePayments(1000, 1);
+        makeGAUs(2);
+        
+        // mark payments as paid
+        for(npe01__OppPayment__c payment: payments) {
+            payment.npe01__Paid__c = true;
+        }
+        update payments;
+
+        System.assert(UTIL_CustomSettingsFacade.getAllocationsSettings().Payment_Allocations_Enabled__c, 
+            UTIL_CustomSettingsFacade.getAllocationsSettings());
+
+        List<Allocation__c> queryAllocations = getAllocationsOrderByAmount();
+        System.assertEquals(1, queryAllocations.size(), 'One allocation should exist.');
+        assertAllocation(queryAllocations[0], 1000, defaultGau.Id, false);
+
+        Decimal origAllocAmount = queryAllocations[0].Amount__c;
+        Decimal origAllocPercent = queryAllocations[0].Percent__c;
+
+        //resetting these manually as a test is considered a single context for static variables
+        List<Allocation__c> alloForInsert = new List<Allocation__c>();
+
+        alloForInsert.add(new Allocation__c(Opportunity__c=opportunity.Id, Percent__c = 50, General_Accounting_Unit__c = gaus[0].Id));
+        alloForInsert.add(new Allocation__c(Opportunity__c=opportunity.Id, Amount__c = 300, General_Accounting_Unit__c = gaus[1].Id));
+        insert alloForInsert;
 
+        queryAllocations = getAllocationsOrderByAmount();
+
+        System.assertEquals(4, queryAllocations.size(), 'Four allocations should exist.');
+
+        //original payment allocations
+        assertAllocation(queryAllocations[0], 200, defaultGau.Id, true);
+        assertAllocation(queryAllocations[1], 300, gaus[1].Id, true);
+        assertAllocation(queryAllocations[2], 500, gaus[0].Id, true);
+        assertAllocation(queryAllocations[3], origAllocAmount, defaultGau.Id, false);
+    }
+
+    /*******************************************************************************************************
+    * @description When attempting to lower or delete an opp allocation for a GAU that has paid payment allocations:
```

Comment:
Do we need to confirm the other case (where there is some paid, and some unpaid), and the allocations (up to the amount of the unpaid payment)?

Reply:
Perhaps, although this wasn't listed specifically in the list of tests Nic left us with... not that we have to anchor ourselves to those specific tests, but in the interest of limiting the scope of each individual test it might be another one I would lobby for an additional test method.

Reply:
Ok.  Why don't we throw these tests in a future Work Item.  Maybe they are good (cutting teeth) Work Items for a new developer.